pkg/routeagent_driver/handlers/healthchecker/healthchecker_test.go

```diff
@@ -0,0 +1,409 @@
+	kubeScheme "k8s.io/client-go/kubernetes/scheme"
+)
+
+const (
+	namespace       = "submariner"
+	remoteClusterID = "west"
+	healthCheckIP1  = "1.1.1.1"
+	healthCheckIP2  = "2.2.2.2"
+	localNodeName   = "nodeName"
+)
+
+var _ = Describe("RouteAgent syncing", func() {
+	t := newTestDriver(true)
+	It("should create a RouteAgent resource", func() {
+		var routeAgent *submarinerv1.RouteAgent
+		Eventually(func() bool {
+			var err error
+			routeAgent, err = t.client.Get(context.TODO(), localNodeName, metav1.GetOptions{})
+			return err == nil && routeAgent != nil
+		}, time.Second*50, time.Millisecond*500).Should(BeTrue())
+		Expect(routeAgent.Name).To(Equal(localNodeName))
+		Expect(routeAgent.Namespace).To(Equal(namespace))
+		Expect(routeAgent.Status.RemoteEndpoints).To(BeEmpty())
+	})
+
+	When("a remote Endpoint is created", func() {
+		It("should add its RemoteEndpoint information to the RouteAgent resource", func() {
+			endpoint1 := t.CreateEndpoint(t.newSubmEndpoint(healthCheckIP1))
+
+			var routeAgent *submarinerv1.RouteAgent
+			Eventually(func() bool {
+				var err error
+				routeAgent, err = t.client.Get(context.TODO(), localNodeName, metav1.GetOptions{})
+				if err == nil && len(routeAgent.Status.RemoteEndpoints) != 0 {
+					return true
+				}
+				return false
+			}, time.Second*50, time.Millisecond*500).Should(BeTrue())
+
+			Expect(routeAgent.Name).To(Equal(localNodeName))
+			Expect(routeAgent.Namespace).To(Equal(namespace))
+
+			remoteEndpoint := routeAgent.Status.RemoteEndpoints[0]
+			Expect(remoteEndpoint.Spec).To(Equal(endpoint1.Spec))
+		})
+	})
+
+	When("a remote Endpoint is updated", func() {
+		It("should update its RemoteEndpoint information in the RouteAgent resource", func() {
+			endpoint1 := t.CreateEndpoint(t.newSubmEndpoint(healthCheckIP1))
+			t.pingerMap[healthCheckIP1].AwaitStart()
+			endpoint1.Spec.Hostname = "newHostName"
+			t.UpdateEndpoint(endpoint1)
+
+			pingerObject, found := t.pingerMap[endpoint1.Spec.HealthCheckIP]
+			Expect(found).To(BeTrue())
+			Expect(pingerObject.GetIP()).To(Equal(healthCheckIP1))
+
+			var routeAgent *submarinerv1.RouteAgent
+			Eventually(func() string {
+				var err error
+				routeAgent, err = t.client.Get(context.TODO(), localNodeName, metav1.GetOptions{})
+				if err == nil && len(routeAgent.Status.RemoteEndpoints) != 0 {
+					return routeAgent.Status.RemoteEndpoints[0].Spec.Hostname
+				}
+				return ""
+			}, time.Second*50, time.Millisecond*500).Should(Equal(endpoint1.Spec.Hostname))
+
+			Expect(routeAgent.Name).To(Equal(localNodeName))
+			Expect(routeAgent.Namespace).To(Equal(namespace))
+
+			remoteEndpoint := routeAgent.Status.RemoteEndpoints[0]
+			Expect(remoteEndpoint.Spec).To(Equal(endpoint1.Spec))
+		})
+	})
+
+	When("a remote Endpoint is deleted", func() {
+		It("should remove its RemoteEndpoint information from the RouteAgent resource", func() {
+			endpoint1 := t.CreateEndpoint(t.newSubmEndpoint(healthCheckIP1))
```

Comment:
Similarly as with update, you can combine delete as well:

`When("a remote Endpoint is created/updated/deleted", func() {`

Reply:
Done.